lexer.html

```diff
@@ -0,0 +1,140 @@
+<link rel="import" href="common.html">
+<script>
+function ShadeyCssLexer() {
+  'use strict';
+  // NOTE(cdata): Generally, tokens are structured as three-valued tuple,
+  // where the first value is the token value, the second value is the token
+  // type and the third value is the cursor index after consuming the raw
+  // source.
+
+  var common = ShadeyCssCommon();
+  var matcher = common.matcher;
+  var tokenType = common.tokenType;
+
+  function Whitespace(source, offset) {
+    var start = offset;
+
+    while (matcher.whitespace.test(source[offset])) {
+      ++offset;
+    }
+
+    return [
+      source.substr(start, offset - start),
+      tokenType.whitespace,
+      offset
+    ];
+  }
+
+  function Boundary(source, offset) {
```

Comment:
Consider separating grouping tokens ( {}, () and [] ) from other punctuation like : and ;